File: Binary Tree Inorder Traversal/Binary Tree Inorder Traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ArrayList<Integer> inorderTraversal(TreeNode root) {
        ArrayList<Integer> list=new ArrayList<Integer>();
        Stack<TreeNode> stack=new Stack<TreeNode>();
        TreeNode temp=root;
        while(!stack.isEmpty()||temp!=null){
            if(temp!=null){
                stack.push(temp);
                temp=temp.left;
            }else{
                temp=stack.pop();
                list.add(temp.val);
                temp=temp.right;
                }
        }
        return list;
    }
}

File: Binary Tree Inorder Traversal/Binary Tree Inorder Traversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of integers
    def inorderTraversal(self, root):
        list=[]
        stack=[]
        temp=root
        while stack or temp:
            if temp:
                stack.append(temp)
                temp=temp.left
            else:
                temp=stack.pop()
                list.append(temp.val)
                temp=temp.right
        return list
